File: 01-backend/spring-boot-library/src/main/java/com/libranova/spring_boot_library/model/Message.java
package com.libranova.spring_boot_library.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Message
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "subject")
    private String subject;

    @Column(name = "inquiry")
    private String inquiry;

    @Column(name = "admin_email")
    private String adminEmail;

    @Column(name = "response")
    private String response;

    @Column(name = "submission_date")
    private String submissionDate;

    @Column(name = "answered")
    private boolean answered;
}

File: 02-frontend/react-library-app/src/models/Message.ts
class Message {
    subject: string;
    inquiry: string;
    id?: number;
    userEmail?: string;
    adminEmail?: string;
    response?: string;
    answered?: boolean;

    constructor(subject: string, inquiry: string) {
        this.subject = subject;
        this.inquiry = inquiry;
    }
}
export default Message;
